doc/sphinx/practical-tools/coqide.rst

```diff
@@ -184,20 +239,61 @@ Notice that these settings are saved in the file ``coqiderc`` in the
 is the value of ``$XDG_CONFIG_HOME`` if this environment variable is
 set and which otherwise is ``$HOME/.config/``.
 
+Key bindings
+~~~~~~~~~~~~
+
+Each menu item in the GUI shows its key binding, if one has been defined,
+on the right-hand side.  Typing the key binding is equivalent to selecting
+the associated item from the menu.
 A GTK+ accelerator keymap is saved under the name ``coqide.keys`` in
-the same ``coq`` subdirectory of the user configuration directory. It
-is not recommended to edit this file manually: to modify a given menu
-shortcut, go to the corresponding menu item without releasing the
-mouse button, press the key you want for the new shortcut, and release
-the mouse button afterwards. If your system does not allow it, you may
-still edit this configuration file by hand, but this is more involved.
+the ``coq`` subdirectory of the user configuration directory,
+e.g. in `~/.config/coq/` on Linux and `C:\Users\<USERNAME>\AppData\Local\coq`
+on Windows. On some systems (not Linux or Windows),
+you can modify the key binding ("accelerator") for a menu entry by
+going to the corresponding menu item without releasing the
+mouse button, pressing the keys you want for the new binding and then releasing
+the mouse button.
+
+Alternatively, you can edit the file directly.  Make sure there are no
+CoqIDE processes running while you edit the file.  (CoqIDE creates or
+overwrites the file when it terminates, which may reorder the lines).
+
+The file contains lines such as:
 
+   ::
+
+     ; (gtk_accel_path "<Actions>/Queries/About" "<Primary><Shift>a")
+     ; (gtk_accel_path "<Actions>/Export/Export to" "")
+     (gtk_accel_path "<Actions>/Edit/Find Next" "F4")
+
+The first line corresponds to the menu item for the Queries/About menu item,
+which was bound by default to Shift-Ctrl-A.  "<Primary>" indicates Ctrl.
+The second line is for a menu item that has no key binding.
+
+Lines that begin with semicolons are comments create by CoqIDE.  CoqIDE uses
+the default binding
+for these items.  To change a key binding, remove the semicolon and set the third item
+in the list as desired, such as in the third line.  Avoid assigning the same binding
+to multiple items.
+
+If the same menu item name appears on multiple lines in the file, the value from the
+last line is used.  This is convenient for copying a group of changes from elsewhere--just
+insert the changes at the end of the file.  The next time CoqIDE terminates, it will
+usually reorder the items.
+
+Some menu entries can be changed as a group from the Edit/Preferences/Shortcuts panel.
+Key bindings that don't appear in the file such as Ctrl-A (Select All) can't be
+changed with this mechanism.
+
+.. todo: list common rebindings?
+
+.. todo: microPG mode?
```

Comment:
I don't particularly remember, but I seem to be under the impression this enables the PG keybindings rather than the default CoqIDE ones.

Reply:
Yeah, I've not found time to experiment with it.  See `Help/Help for uPG mode` and `Preferences/Editor`.  On the former, I wonder what key `Meta` corresponds to.  Also what happens to the accelerator settings it clashes with.